app/page/rescuePlanPage.js

```diff
@@ -0,0 +1,143 @@
+import hmUI from '@zos/ui';
+import { push } from '@zos/router';
+import {
+	TOGGLE_OPTIONS,
+	BACK_BUTTON,
+} from 'zosLoader:./permissionsPage.[pf].layout.js';
+import { LocalStorage } from '@zos/storage'; // Import LocalStorage only once
+
+console.log('Navigated to the permissions page');
+
+Page({
+	state: {
+		message: 'Permissions Page',
+		sleepScore: true,
+		startEndTime: true,
+		deepSleepTime: true,
+		totalSleepTime: true,
+		wakeStage: true,
+		remStage: true,
+		lightStage: true,
+		deepStage: true,
+	},
+
+	onInit() {
+		console.log('Permissions Page onInit invoked');
+		this.localStorage = new LocalStorage(); // Initialize LocalStorage once
+		this.restoreToggleStates(); // Restore saved states when the page is initialized
+	},
+
+	build() {
+		console.log('Permissions Page Build Called');
+
+		// Display the main title
+		hmUI.createWidget(hmUI.widget.TEXT, {
```

Comment:
same reused code from the other file, so it introduces more ways to mess up if this needs to change, so it's easier to just extract to a single source of truth. don't need to do this now if you don't want to, you can just include it later